src/main/java/com/uber/cadence/worker/Worker.java

```diff
@@ -64,13 +71,26 @@
    * @param taskList task list name worker uses to poll. It uses this name for both decision and
    *     activity task list polls.
    * @param options Options (like {@link DataConverter} override) for configuring worker.
+   * @param stickyTaskListName
    */
-  private Worker(IWorkflowService service, String domain, String taskList, WorkerOptions options) {
+  private Worker(
+      IWorkflowService service,
+      String domain,
+      String taskList,
+      WorkerOptions options,
+      DeciderCache cache,
+      String stickyTaskListName,
+      Duration stickyDecisionScheduleToStartTimeout,
```

Comment:
yes, but for the most part it's expected that worker wouldn't need them since these options will be to instantiate/set objects that are outside the worker's scope (cache and poller). I can refactor then if there's a pattern of having to pass them in.